### ISSUE TITLE
Does this project still being maintained?

### ISSUE THREAD

**Issue by timqian** [sloisel/numeric#72]
Last commit is back to 2012.

**Comment by victorsosa** [comment]
it seems like

**Comment by angeris** [comment]
I emailed the repo owner not too long ago and he seems to be relatively busy, though interested in people still submitting patches. He also mentioned that the library goes through a very particular test system before any revision, which is why it wasn't so easy for him to be able to keep up with it.

Anyways, that's about all I have in terms of this. I would be interested in working on the convex solver for a similar library, if anyone is willing to either fork this project for a full active branch, or start a new project.

**Comment by Skylion007** [comment]
Yeah, it really concerns me that basic issues such as #66 have not be resolved. He may be interested in people submitting patches, but he has not accepted any it seems.

**Comment by reptillicus** [comment]
I don't believe that it is. Someone who has time should fork it and take
over maintanence.

**Comment by Skylion007** [comment]
Yeah I just emailed the repo owner to ask if he'd be willing to add a collaborator to the repository to help with merging commits.

**Comment by reptillicus** [comment]
I'd be willing to help out with maintanence to some degree. I'm somewhat
busy for the next few months but could chip in.

**Comment by angeris** [comment]
If anyone is still up for forking the project and continuing a main development branch, I'd still be interested in some work on the general optimization solvers and such.

**Comment by Skylion007** [comment]
Okay, I've waited two weeks and haven't received a response. It might just be easiest if we get an official fork going. Anyone else in seeing an updated v2 version of this repo?

**Comment by angeris** [comment]
Yeah, I don't think there's an updated version with any major revisions, to be honest, just a bunch of pull requests/small patches. So, go ahead and fork it, I think we'd all be on board with it and just start contributing.

**Comment by beehorf** [comment]
I think the project should be maintained and improved in some way. I am also intereseted in and can help as much i can. The framework(a numeric enviroment) i work on highly depend on numeric.js

**Comment by angeris** [comment]
Okay, so, could somebody who is willing to be active on the project just branch it such that we can all contribute?

**Comment by victorsosa** [comment]
I can do that :+1: 

https://github.com/peoplewareDo/numeric

Send you patches and let works on it

**Comment by Skylion007** [comment]
I was waiting for the repo owner to respond by the end of the week, but I suppose this could work too. Shouldn't we do a "hard" fork instead of a soft fork though? I know my behavior is just click through to the original repo on soft-fork. Also maybe we should list it under a Github organization for organization's sake in case this ever happens again?

**Comment by Skylion007** [comment]
Also, we probably want to rebrand it to note that behavior might significantly diverge from the original such as numeric2.js.

**Comment by victorsosa** [comment]
I put it under an Organization so, just send the patch with the rebrand to the new repo. Let's work.

**Comment by Skylion007** [comment]
I still think it should be a hard fork though, instead of being a soft fork on Github. Not sure how to convert it without deleting the repo though. (The way to do a hard fork is to clone the repo, then change the repo remote location and push it to the new location, that way it makes a full copy of the git history instead of doing whatever wizardy deduplication stuff Github is doing behind the scenes. 

PS: Add me to the organization.

**Comment by angeris** [comment]
Yeah, that sounds good... I think the author wouldn't mind too much and worst comes to worst, we could just consolidate with the author at the end.

P.S. It would be great if I could also be added to the Organization.

**Comment by sloisel** [comment]
Hi,

Of course you guys can fork if you want!

The main problem that I have is that, whenever I make a change, I must run the entire test suite in a variety of browsers. I've set it up on my mac to run in Chrome, Firefox, as well as on IE in a Windows Parallels machine; to me this seemed like the minimal coverage necessary. I have found that testing on all browsers is very important because there's often something that will fail in only one of them.

To do this, I use Selenium, and this is where it gets hairy. Every time I want to release, I invariably have to update Selenium and all plugins, across two operating systems, and also often I have to upgrade my scripts that do the communication between mac and Windows for this testing.

I only release when all the tests are green. It should look like this:
http://www.numericjs.com/report.html

Fancy people with money have "continuous deployment" or whatever, a farm of servers that automatically test in a variety of browsers, but I don't have this and I don't want to sysadmin it either!

**Comment by reptillicus** [comment]
Sauce labs if free for open source projects, could look into that. 
https://saucelabs.com/opensauce/

**Comment by moleike** [comment]
@sloisel if that is the problem, you could also check https://ci.testling.com/

**Comment by timqian** [comment]
I am doing a refactoring of this project recently. ( https://github.com/timqian/mathlab )
The aims includes:
1. Make the pointwise functions also suitable for Complex and Sparse matrix.
2. Not using function constructor to generate functions
3. Separate the package into small files so that people can require functions only when they need. 
4. Unit test using mocha
5. Rename the T to Complex and {x: ..., y: ...} to {re: ..., im:...}
6. Make sparse matrix to be an Constructor
7. add more functions
8. ...

Please let me know your thoughts about it.

**Comment by mitar** [comment]
Yes, you can configure Sauce labs for automatic GitHub integrated CI testing on real browsers. Each pull request could then be automatically checked against a test suite. So you would not have to run anything locally. It is free for open source projects.

**Comment by angeris** [comment]
@timqian Once you get the CI testing/refactoring done and such, I'd be happy to contribute to debugging some of the optimization methods.

**Comment by Skylion007** [comment]
@mitar Want to help setup the integrated CI testing on the peopledoware fork? I want to get the automated testing setup on at least one fork so we can get the benchmarking setup on at least one fork.

**Comment by yig** [comment]
Anyone forking numericjs should work from Daniel Holden's fork ( @orangeduck ), which added NumPy-style broadcasting and other useful stuff. He did good work. He even updated the documentation page.

Here is his repository:
    https://github.com/orangeduck/numeric

Here is his pull request:
    https://github.com/sloisel/numeric/pull/56

Here is his announcement on the Google Group:
    https://groups.google.com/forum/#!topic/numericjs/giRexhL3ydg

**Comment by mitar** [comment]
I made #76 with Travis CI tests.

**Comment by mitar** [comment]
You can see example run here: https://travis-ci.org/mitar/numeric/builds/180668112

**Comment by mitar** [comment]
@sloisel: Do you care giving commit bits to few people here? I think there are quite few who would be willing to help maintain it.

**Comment by mitar** [comment]
Now we test on multiple node version using Travis CI, and on Firefox and Chrome using Circle CI, and IE using Circle CI + Sauce Labs. See example: https://circleci.com/gh/mitar/numeric/17

**Comment by quantuminformation** [comment]
@sloisel may I advise that you set up a Patreon project for this? I'm sure the people here who make money from this project will be happy to contribute.

**Comment by mitar** [comment]
I would just advice to open maintenanceship. This project is not being maintained for 5 years now.
